### PR TITLE
[cookbook] Add annotations block and fix regex

### PR DESCRIPTION
**Pull request by petk**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | all |
| Fixed tickets | N/A |

### CODE REVIEW

**Comment by javiereguiluz** [comment]
:+1: thanks for this improvement @peterkokot  (I find the code of the `removeTrailingSlashAction()` method too "complex" for this trivial case ... but I can't come up with a better proposal).

File: cookbook/routing/redirect_trailing_slash.rst [hunk 1]
```diff
@@ -37,13 +37,38 @@ system, as explained below:
...
+             *     requirements={"url" = ".*\/$"}, methods={"GET"})
+             */
+            public function removeTrailingSlashAction(Request $request)
+            {
+                $pathInfo = $request->getPathInfo();
+                $requestUri = $request->getRequestUri();
+
+                $url = str_replace($pathInfo, rtrim($pathInfo, ' /'), $requestUri);
+
+                return $this->redirect($url, 301);
```
Comment (xabbuh) [inline cookbook/routing/redirect_trailing_slash.rst]: We already have the complete code above the configurations so I would reduce this to the bare minimum like this:

```rst
.. configuration-block::

    .. code-block:: php+annotations::

        // src/AppBundle/Controller/RedirectingController.php
        
        // ...
        class RedirectingController extends Controller
        {
            /**
             * @Route("/{url}", name="remove_trailing_slash",
             *     requirements={"url" = ".*\/$"}, methods={"GET"})
             */
            public function removeTrailingSlashAction(Request $request)
            {
                // ...
            }
        }
```

**Comment by xabbuh** [comment]
Do we really need to escape the slash?

**Comment by WouterJ** [comment]
Thank you @peterkokot! I've merged your PR into 2.3.

It turns out the slash doesn't have to be escaped in the regex, so I reverted that change in d3744cb (I kept it for the annotations, as it's required there).

**Comment by taylankasap** [comment]
How are you supposed to

> put this route last in your system

when used with annotations?

**Pull request closed without merging**